src/IsomorphicRequest.ts

```diff
@@ -0,0 +1,57 @@
+import { Headers } from 'headers-polyfill/lib'
+import { decodeBuffer } from './utils/bufferUtils'
+import { uuidv4 } from './utils/uuid'
+
+export interface RequestInit {
+  method?: string
+  headers?: Record<string, string | string[]> | Headers
+  credentials?: RequestCredentials
+  body?: ArrayBuffer
+}
+
+export class IsomorphicRequest {
+  public id: string
+  public readonly url: URL
+  public readonly method: string
+  public readonly headers: Headers
+  public readonly credentials: RequestCredentials
+  public readonly body: ArrayBuffer
```

Comment:
Any reason to keep this public?

Reply:
No. Following our discussion in Discord, let's make `body` private because we currently don't comply with the spec in this regard. We are not returning `body` as a readable stream.